engines/director/debugger.cpp

```diff
@@ -558,8 +605,19 @@ bool Debugger::cmdDisasm(int argc, const char **argv) {
 				debugPrintf("%s\n", lingo->formatFunctionBody(ctx->_functionHandlers[funcName]).c_str());
 				return true;
 			}
+			if (sharedCast->_lingoArchive->factoryContexts.contains(scriptId)) {
```

Comment:
This block looks to be a duplicate of the one at line 588. Could both blocks be moved in a reusable function?

Reply:
I've removed the main source of duplication (having to run everything for both the cast and the sharedCast). Most of what's left is looping through the different places script archives are stored and calling formatFunctionBody on them; my preference is to keep the logic for doing so as flat as possible so it's easier to read.